conda/core/solve/state.py

```diff
@@ -0,0 +1,717 @@
+                else:
+                    reason = "Pinning Python to match installed version"
+                    version = get_major_minor_version(record.version) + ".*"
+                    spec = MatchSpec(spec, version=version)
+
+                # There's a chance the selected version results in a conflict -- detect and report?
+                specs = (spec, ) + tuple(sis.requested.values())
+                if sis.get_conflicting_specs(specs, sis.requested.values()):
+                    if not sis.installing:  # TODO: repodata checks?
+                        # raises a hopefully helpful error message
+                        sis.find_conflicts(specs)  # this might call the solver -- remove?
+                    else:
+                        # oops, no message?
+                        raise RawStrUnsatisfiableError("Couldn't find a Python version that does not conflict...")
+
+                self.specs.set("python", spec, reason=reason)
+
+
+        ### Offline and aggressive updates ###
+
+        # For the aggressive_update_packages configuration parameter, we strip any target
+        # that's been set.
+
+        if not context.offline:
+            for name, spec in sis.aggressive_updates:
+                if name in self.specs:
+                    self.specs.set(name, spec, reason="Aggressive updates relaxation")
+
+        ### User requested specs ###
+
+        # add in explicitly requested specs from specs_to_add
+        # this overrides any name-matching spec already in the spec map
+
+        for name, spec in sis.requested:
+            if name not in pin_overrides:
+                self.specs.set(name, spec, reason="Explicitly requested by user")
+
+        ### Conda pinning ###
+
+        # As a business rule, we never want to downgrade conda below the current version,
+        # unless that's requested explicitly by the user (which we actively discourage).
+
+        if "conda" in self.specs and paths_equal(sis.prefix, context.conda_prefix) and "conda" in sis.installed:
+            record = sis.installed["conda"]
+            spec = self.specs["conda"]
+            required_version = f">={record.version}"
+            if not spec.get("version"):
+                spec = MatchSpec(spec, version=required_version)
+                reason = "Pinning conda with version greater than currently installed"
+            if context.auto_update_conda and "conda" not in sis.requested:
+                spec = MatchSpec("conda", version=required_version, target=None)
+                reason = "Pinning conda with version greater than currently installed, auto update"
+            self.specs.set("conda", spec, reason=reason)
+
+        ### Extra logic ###
+        # this is where we are adding workarounds for mamba difference's in behavior, which might not belong
+        # here as they are solver specific
+
+        # next step -> .prepare_for_solve()
+
+
+    def prepare_for_remove(self) -> dict[str, MatchSpec]:
+        pass
```

Comment:
What is this used for again?

Reply:
So far I only implemented the state collection for `conda install/update/create`; this one would be for `conda remove`.